File: LeetCode/LeetCode Practice Problems/1002- Find Common Characters.cpp
class Solution {
public:
    vector<string> commonChars(vector<string>& A) {
        vector<string> ans;
        int n = A.size();
        vector<vector<int>> cnt(n,vector<int>(26,0));
        
        for(int i=0;i<n;i++) {
            for(auto j:A[i])
                cnt[i][j-'a']++;
        }
        
        for(int i=0;i<26;i++) {
            int mn = cnt[0][i];
            for(int j=0;j<n;j++) {
                mn = min(mn , cnt[j][i]);
            }
            char ch = i+'a';
            string s;
            s+=ch;
            while(mn--) {
                ans.push_back(s);
            }
        }
        
        return ans;
        
    }
};

File: LeetCode/LeetCode Practice Problems/1002-Find Common Characters.java
class Solution {
    public List<String> commonChars(String[] words) {
        int cnt[] = new int[26];
        Arrays.fill(cnt , Integer.MAX_VALUE);
        for(String word: words) {
            int f[] = new int[26];
            for(char c: word.toCharArray()) {
                f[c - 'a']++;
            }
            for(int i=0;i<26;i++) {
                cnt[i] = Math.min(cnt[i] , f[i]);
            }
        }
        List<String> ans = new ArrayList();
        for(int i=0;i<26;i++) {
            while(cnt[i] > 0) {
                ans.add((char)(i + 'a') + "");
                cnt[i]--;
            }
        }
        return ans;
    }
}
